### ISSUE TITLE
[Exception: The attribute definition cannot be added to the collection] when loading dxf file

### ISSUE THREAD

**Issue by eplesse** [haplokuon/netDxf#207]
Hello,
First congrats for your great lib which I have been using for a year now in order to instantiate autocad objects into 3d UNITY app.

I recently ran into this exception (using latest 2.4.0 release) when loading a dxf file (converted from a dwg file with AUTOCAD Architecture 2021).
Commenting lines 197,198 in "AttributDefinitionDictionnary.cs" helped me to bypass this exception and enabled me to load objects I needed. But of course this is no long-term solution.

I can provide you with the dxf file (which opens perfectly in AUTOCAD) but for confidentiality reasons, I cannot upload it here.

thanks. Emmanuel

**Comment by haplokuon** [comment]
You can send me the DXF directly using the email you can find at the header of the source code files if you need to keep it private.

**Comment by haplokuon** [comment]
I checked your file and...

This is an oversight on my part. Commenting those lines will avoid other conditions where an attribute definition cannot be added to the collection. Instead go to the Block class and in line 751, inside the "AttributeDefinitions_BeforeAddItem" method, delete the whole else if block:
```c#
else if (this.Name.StartsWith(DefaultPaperSpaceName)) // paper space blocks do not contain attribute definitions
{
	e.Cancel = true;
}
```

Paper space blocks can indeed contain attribute definitions, it is uncommon but there is nothing in the DXF format that forbids it.

**Comment by eplesse** [comment]
Ok. This get rids of the problem and I can load all objects correctly. Thanks

**Comment by haplokuon** [comment]
Resolved in 2.4.1
